packages/manager/modules/request-tagger/src/factory.js

```diff
@@ -0,0 +1,47 @@
+import find from 'lodash/find';
+import omit from 'lodash/omit';
+import startsWith from 'lodash/startsWith';
+
+import {
+  HEADER_NAVIGATION_ID,
+  HEADER_REQUEST_ID,
+  HEADER_PAGE,
+  ROUTES_PREFIX,
+  ROUTES_HEADERS_OVERRIDE,
+} from './constants';
+
+export default /* @ngInject */ () => {
+  let requestIndex = 0;
+  return {
+    request: (config) => {
+      let taggedConfig;
+
+      if (find(ROUTES_PREFIX, route => startsWith(config.url, route))) {
+        requestIndex += 1;
+        const overridenHeaders = find(
+          ROUTES_HEADERS_OVERRIDE,
+          (value, route) => new RegExp(route).test(config.url),
+        ) || {};
+
+        taggedConfig = {
+          ...config,
+          headers: {
+            ...config.headers,
+            [HEADER_REQUEST_ID]: `${Date.now()}-${requestIndex}`,
+            ...overridenHeaders,
+          },
+        };
+      } else {
+        taggedConfig = {
+          ...config,
+          headers: omit(config.headers, [
+            HEADER_NAVIGATION_ID,
+            HEADER_REQUEST_ID,
+            HEADER_PAGE,
+          ]),
+        };
+      }
+      return taggedConfig;
```

Comment:
You could `return` directly and avoid the `else` 
```suggestion
      if (find(ROUTES_PREFIX, route => startsWith(config.url, route))) {
        requestIndex += 1;
        const overridenHeaders = find(
          ROUTES_HEADERS_OVERRIDE,
          (value, route) => new RegExp(route).test(config.url),
        ) || {};

         return {
          ...config,
          headers: {
            ...config.headers,
            [HEADER_REQUEST_ID]: `${Date.now()}-${requestIndex}`,
            ...overridenHeaders,
          },
        }
        
        return {
          ...config,
          headers: omit(config.headers, [
            HEADER_NAVIGATION_ID,
            HEADER_REQUEST_ID,
            HEADER_PAGE,
          ]),
        };
      
```